boudicca.base/common-model/src/main/kotlin/base/boudicca/model/structured/StructuredEventBuilder.kt

```diff
@@ -0,0 +1,26 @@
+package base.boudicca.model.structured
+
+import java.time.OffsetDateTime
+
+class StructuredEventBuilder(
+    private val name: String,
+    private val startDate: OffsetDateTime,
+    data: Map<Key, String> = emptyMap()
+) : AbstractStructuredBuilder<StructuredEvent, StructuredEventBuilder>(data.toMutableMap()) {
+
+    fun withData(name: String, init: DataBuilder.() -> Unit) {
+        val builder = DataBuilder(name)
+        builder.init()
+        val dataEntries = builder.build()
+
+        this.data.putAll(dataEntries)
+    }
+
```

Comment:
why did you remove the copy()/withName()/withStartDate() methods?

Reply:
Why do we need them?
Is there really a case where event name and start date are unknown at event creation time?

This way we can get rid of the nullability, which resolves a lot of headaches.

Similarly I do not see the use case for a copy operation?

Reply:
you could just look at the usages of the methods :P in our case collectors like BrucknerhausCollector benefit from that. they have the same event at different starttimes, so they create the full event, and then iterate over all starttimes, copy the builder, set the time and call .build()

Reply:
Hmm. I was looking for usages, but couldn't find any. thats why i removed the. maybe i have to look again.

Reply:
while the change is fine with me "the new way should also have slightly improved memory performance." is not true since now you parse urls and execute CSS selectors in a loop instead of once :P